### PR TITLE
[IntegrationSpecs] Use developer tools when available

### PR DESCRIPTION
**Pull request by segiddins**
\c @mrackwitz

### CODE REVIEW

**Comment by mrackwitz** [comment]
Do we really want that? We're lately very often on different versions of the toolchain and it would be better if we use similar versions of the tools, which is perhaps more likely if we use either the version of Xcode on the default path or the version shipped with the OS.

**Comment by segiddins** [comment]
That's why this is using what apples ships with Xcode, since I have a never version of git that breaks the integration specs.

**Comment by segiddins** [comment]
I'm merging this unless anyone objects

File: spec/integration.rb [hunk 1]
```diff
@@ -94,6 +94,11 @@
   Process.wait(spawn('which hg', :err => :out, :out => '/dev/null'))
   has_mercurial = $?.success?
 
+  if (developer_bin = `xcode-select -p 2>/dev/null`.strip) && $?.success?
+    developer_bin = Pathname(developer_bin) + 'usr/bin'
+    ENV['PATH'] = "#{developer_bin}#{File::PATH_SEPARATOR}#{ENV['PATH']}"
```
Comment (mrackwitz) [inline spec/integration.rb]: As this has global side-effects, I would also put it in the global scope at the beginning of the file.

Comment (segiddins) [inline spec/integration.rb]: :+1:

**Pull request merged**